File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        	int ans = 0;
	set<int> s ;

	for (int i = 0 ; i < (int)nums.size() ; i ++ ){
		s.insert(nums[i]);
	}

	for (auto i : s ){
		if (s.find(i-1) == s.end()){
			int length = 1 ;
			while(s.find(length + i) != s.end()){
				length ++;
			}
			ans = max(ans , length);
		}
	}


	return ans;
    }
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
            if len(nums) == 0:
                return 0

            nums = set(nums)
            nums = list(nums)
            nums.sort()

            ans = 0
            cnt = 0
            for i in range(len(nums)-1):
                if nums[i]+1 == nums[i+1]:
                    cnt += 1
                else:
                    ans = max(ans ,cnt)
                    cnt = 0

            ans = max(ans , cnt)
            return ans + 1
